### ISSUE TITLE
Thoughts

### ISSUE THREAD

**Issue by mahtoid** [ErikMekkes/DiscordChatExporterPy#4]
TL;DR: If YOUR personal goal is to have long standing chat exports, then I feel as if you've achieved this. However, there are many faults with storing files locally from breaching Discords TOS, personal security and just sanity to do so - which I personally don't agree with, and the main lib won't ever support. 

---

[**`local_download()`**](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-82a555ac3e707c633d7ff9ffe8f3d0c0b9a440b3ab875ed4c541832353233306R20-R62)
Storing files locally is something I disagree with. There are so many exploits and mallicious content it could be - especially coming from Discord. I do recognise that you've only done this to embeds (for now) - but it's still something to be wary off and not something I'd like to advocate.

---

[**extensions = [ ... ]**](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-82a555ac3e707c633d7ff9ffe8f3d0c0b9a440b3ab875ed4c541832353233306R30-R42)
This could probably be turned in to a dict, to avoid for loops.

---

[**import requests**](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-82a555ac3e707c633d7ff9ffe8f3d0c0b9a440b3ab875ed4c541832353233306R2)
`requests` is a bad module to use, `aiohttp` is much better - allows for asynchronous functions and a module the lib is already using. If you check out [this funky issue](https://github.com/mahtoid/DiscordChatExporterPy/issues/15) you'll see the negative effect of using `requests` in the lib. 

---

[**`proxy_url` elif `url`**](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-82a555ac3e707c633d7ff9ffe8f3d0c0b9a440b3ab875ed4c541832353233306R223-R244)
If there is no `proxy_url`, there is no `url`. 

The `proxy_url` is a cached version of the assets URL - which will still exists briefly in the case of the asset being deleted from the channel. [See more](https://discordpy.readthedocs.io/en/stable/api.html#discord.Attachment.proxy_url)

Knowing the above, it'll be easier to clean up [`download_embeds()`](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-82a555ac3e707c633d7ff9ffe8f3d0c0b9a440b3ab875ed4c541832353233306R65-R126) so you'd have a lot less code duplication. 

---

[**Missing `</div>`'s**](https://github.com/ErikMekkes/DiscordChatExporterPy/commit/1349429b2a947bd216577209e4727dab7c85abb4#diff-e0acb89e655d54252cb4cc041a11214b231a9b81e4657f0762cfb3479bbe9f11R372-R373)
Although very disgusting, this is intentional. Adding those 2 `</div>` tags will signify the end of the users message, but that isn't always what we want when we want to group a bunch of messages together.

---

I hope what I have said hasn't knocked you in a negative manner, it wasn't meant to. A lot of the blockers I see with this is remaining inside Discords Guidelines, along with reducing security threats to local machines. 

Of course if you choose to run your own fork and keep it going, then feel free - that is the beauty of open-source. However, personally I'd prefer to stick within the lines - so that's why it won't be supported within the main lib. 

Beautifying HTML would be the only thing I'd be able to support due to the fact it doesn't breach Discord ToS or have a threat to security. Please excuse me as I wear my tin-foil hat.

I really appreciate and enjoy seeing peoples additions to the lib, so if you make any further changes - please let me know and I'd be happy to review them.
